komfydb/execution/delete.cc

```diff
@@ -0,0 +1,81 @@
+#include "komfydb/execution/delete.h"
+
+#include "komfydb/common/int_field.h"
+#include "komfydb/common/type.h"
+#include "komfydb/storage/buffer_pool.h"
+#include "komfydb/storage/page_id.h"
+#include "komfydb/storage/record.h"
+#include "komfydb/storage/record_id.h"
+#include "komfydb/utils/status_macros.h"
+
+namespace {
+
+using komfydb::common::IntField;
+using komfydb::storage::BufferPool;
+using komfydb::storage::PageId;
+using komfydb::storage::Record;
+using komfydb::storage::RecordId;
+using komfydb::transaction::TransactionId;
+
+};  // namespace
+
+namespace komfydb::execution {
+
+Delete::Delete(std::unique_ptr<OpIterator> child, uint32_t table_id,
+               TupleDesc tuple_desc, std::shared_ptr<BufferPool> bufferpool,
+               TransactionId tid)
+    : OpIterator(tuple_desc, *child->GetFieldsTableAliases()),
+      child(std::move(child)),
+      bufferpool(std::move(bufferpool)),
+      tid(tid),
+      table_id(table_id) {}
+
+absl::StatusOr<std::unique_ptr<Delete>> Delete::Create(
+    std::unique_ptr<OpIterator> child, uint32_t table_id,
+    std::shared_ptr<BufferPool> bufferpool, TransactionId tid) {
+  std::vector<Type> types = {Type::INT};
+  std::vector<std::string> fields = {"deleted_tuples"};
+  return std::unique_ptr<Delete>(new Delete(std::move(child), table_id,
```

Comment:
shouldn't we check if child is from a single table? E.g. return error if this operator joins tuples. If not here, then in bufferpool for sure.

Reply:
Right now `page->RemoveRecord()` (called by bufferpool) will return error if it ecounters nonsensical RecordId.

Reply:
Why not return it immediately?

Reply:
How to know if child joins tuples without getting Records from it?